src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Snapshot/AggregateDependenciesSnapshotProvider.cs

```diff
@@ -91,20 +101,17 @@ public IDependenciesSnapshot GetSnapshot(string projectFilePath)
         {
             Requires.NotNullOrEmpty(projectFilePath, nameof(projectFilePath));
 
-            lock (_snapshotProviders)
+            if (!_snapshotProviderByProjectPath.TryGetValue(projectFilePath, out IDependenciesSnapshotProvider snapshotProvider))
             {
-                if (!_snapshotProviders.TryGetValue(projectFilePath, out IDependenciesSnapshotProvider snapshotProvider))
-                {
-                    snapshotProvider = _projectExportProvider.GetExport<IDependenciesSnapshotProvider>(projectFilePath);
+                snapshotProvider = _projectExportProvider.GetExport<IDependenciesSnapshotProvider>(projectFilePath);
 
-                    if (snapshotProvider != null)
-                    {
-                        RegisterSnapshotProvider(snapshotProvider);
-                    }
+                if (snapshotProvider != null)
+                {
+                    RegisterSnapshotProvider(snapshotProvider);
```

Comment:
❗️ If `GetSnapshot` is called in parallel, could this result in multiple event handlers getting registered?